packages/node-app/src/controllers/giveaway.controller.ts

```diff
@@ -41,7 +31,37 @@ export const listGiveaways = async (req: Request, res: Response) => {
       )
       .lean();
 
-    giveaways = giveaways.map((giveaway) => ({
+    const active = req.query.active === 'true' ? true : false;
+
+    const activeGiveaways = giveaways?.filter((g) => {
+      const now = new Date();
+      const giveawayStartDate = new Date(g.startTime);
+      const giveawayEndDate = new Date(g.endTime);
```

Comment:
I believe now is not being used and there is no need to have giveawayStartDate and giveawayEndDate as well